www/js/survey/enketo/answer.js

```diff
@@ -35,23 +36,37 @@ angular.module('emission.survey.enketo.answer', [
    * }}
    */
 
-  const ENKETO_SURVEY_CONFIG_PATH = 'json/enketoSurveyConfig.json';
   const LABEL_FUNCTIONS = {
-    TripConfirmSurvey: (xmlDoc) => {
-      const modeStr = _getAnswerByTagName(xmlDoc, 'travel_mode');
-      const purposeStr = _getAnswerByTagName(xmlDoc, 'destination_purpose');
+    UseLabelTemplate: (xmlDoc, name) => {
 
-      if (modeStr.includes('trip_not_valid') || purposeStr.includes('trip_not_valid')) {
-        return 'Trip not valid';
-      }
+      return _lazyLoadConfig().then(configSurveys => {
 
-      const purposes = purposeStr.split(' ').length;
-      const modes = modeStr.split(' ').length;
-      return `${purposes} purpose${purposes > 1 ? 's': ''}, ${modes} mode${modes > 1 ? 's': ''}`;
-    },
-    UserProfileSurvey: (xmlDoc) => {
-      return 'Answered';
-    },
+        const config = configSurveys[name]; // config for this survey
+        const lang = $translate.use();
+        const labelTemplate = config.labelTemplate?.[lang];
+
+        if (!labelTemplate) return "Answered"; // no template given in config
+        if (!config.labelVars) return labelTemplate; // if no vars given, nothing to interpolate,
+        // so we return the unaltered template
+
+        // gather vars that will be interpolated into the template according to the survey config
+        const labelVars = {}
+        for (lblVar in config.labelVars) {
+          const fieldName = config.labelVars[lblVar].key;
+          let fieldStr = _getAnswerByTagName(xmlDoc, fieldName);
+          if (fieldStr == '<null>') fieldStr = null;
+          if (config.labelVars[lblVar].type == 'length') {
+            const fieldMatches = fieldStr?.split(' ');
+            labelVars[lblVar] = fieldMatches?.length || 0;
+          } else {
+            labelVars[lblVar] = fieldStr || '';
+          }
```

Comment:
Not yet. we left it open for other options but our use cases right now are just 'length'

I think we could implement a 'list', which puts the specific choices, not categories, in a comma separated list
However, I don't think we can use MessageFormat to format a list. [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) will do it

Reply:
I think that a comma separated list will be too verbose. If we do want to support a `summary` option, I think it would be `first item (in the current language) +(length-1)` e.g. `(eating +2)` or `(சாப்பாடு + 2)`

Reply:
I am trying to implement this but I have a few questions about the `summary` option:

1. What fields do we want to be summarized? We can either do:
| If the results were `Domestic activities: 1 response (cleaning), Personal Care: 2 responses (doing_sport & sleeping), Recreation activities: 0 responses, etc. |`
| :-----: |
| "Domestic Activities + 1" |
| **OR** "doing_sport + 1" |

2. How do we want the labelTemplate to look? My thoughts:

For "length", we currently interpolate labelVars onto labelTemplate:
```
labelVars = {ral: 0, da: 1, pca: 2}
labelTemplate = { pca, plural, =0 {} other {# Personal Care, }}{ da, plural, =0 {} other {# Domestic, } }{ ral, plural, =0 {} other {# Recreation/Leisure, } }
```

Gives: `"2 Personal Care, 1 Domestic Activities"`

Now _(if we want to display like "Domestic Activities + 1"),_ I believe we need to change labelVars to keep track of how many responses each question has (`all: 3`), but labelTemplate must be changed somehow to display this correctly, my first idea is nesting:
```
labelVars = {all: 3, ral: 0, da: 1, pca: 2}
labelTemplate = { pca, plural, =0 { da, plural, =0 { ral, plural, =0 { No responses } other { Recreation and Leisure } } other { Domestic Activities + {all}-1 } } other { Personal Care + {all}-1 }}
```

Reply:
**I don't think we should implement the summary option right now**. We should wait until we have an actual use case of somebody who wants a different type of label and implement it then.

But if/when we decide to implement it, I think we would want `"doing_sport + 1"`

Reply:
for the record, the survey spec does in fact have multiple languages so the label is `How_old_are_you:label` without any ? etc. the spanish version is ¿Cuántos años tienes? and the english version is How old are you? So it seems like we should be able to find the string to display from the result (and presumably the result has the label `How_old_are_you` instead of any particular formatted language).

```
<fieldset class=\"question simple-select \"><fieldset><legend><span lang=\"es\" class=\"question-label \" data-itext-id=\"/aSfdnWs9LE6q8YEF7u9n85/group_hg4zz25/How_old_are_you:label\">¿Cuántos años tienes?</span><span lang=\"en\" class=\"question-label active\" data-itext-id=\"/aSfdnWs9LE6q8YEF7u9n85/group_hg4zz25/How_old_are_you:label\">How old are you?</span>
```